examples/index.md

```diff
@@ -389,127 +342,61 @@ Content-Type: application/vnd.api+json
 {
   "errors": [
     {
-      "title":  "Missing Data Member",
-      "source": { "pointer": "/data" }
+      "source": { "pointer": "/" },
+      "detail":  "Missing `data` Member at document's top level."
     }
   ]
 }
 ```
 
-Notice that the `source` indicates the error was causes not by a missing attribute, but
-by missing `data`.
+It uses `source` to point to the top-level of the document (`"/"`).
+(Pointing to `"/data` would be invalid because the request document did
+not have a value at `"/data"`, and `source` is always given with reference
+to the request document.)
 
-### Responses to failed resource request:
+Note that if the server cannot parse the request as valid JSON, including
+`source` doesn't make sense (because there's no JSON document for `source` to
+refer to). Here's how the server might respond to an invalid JSON document:
 
-Response with one error object, with parameter error:
-
-```http
-GET /api/posts/1
 ```
-
-```http
-HTTP/1.1 400 Bad Request
-Content-Type: application/vnd.api+json
-
 {
-  "errors": [
-    {
-      "title":  "Missing Query Parameter",
-      "source": { "parameter": "auth_key" },
-      "detail": "The required parameter, auth_key, is missing."
-    }
-  ]
+  "errors": [{
+    "status": "400",
+    "detail": "JSON parse error - Expecting property name at line 1 column 2 (char 1)."
+  }]
 }
 ```
 
-Notice that the `source` indicates the URI query parameter which caused the error.
-This example with a missing `auth_key` parameter was chosen over an example with an invalid
-URI query parameter, since it may make sense to ignore invalid parameters in some cases, and
-have the request succeed rather than fail. As described in the [Bulk
-extension](http://jsonapi.org/extensions/bulk/#bulk-operations):
-
-> ...a request MUST completely succeed or fail (in a single "transaction").
-> ...any request that involves multiple operations MUST only succeed if all operations are performed
-successfully
-
-Examples of invalid parameters may be `?page[prev]=` (parameter present, but has no value),
-`?include=auther` (invalid parameter value, perhaps a typo), `?felds[author]=`,
-(invalid parameter key, perhaps a typo), `?redirect_to=http%3A%2F%2Fwww.owasp.org` (invalid
-parameter, in this case, a phishing attack), etc.
-
-### Using JSON API extensions
```

Comment:
Update: just saw that you suggested putting the extension errors onto the respective extension pages. That could work, perhaps, as an separate PR, assuming they can be integrated unobtrusively (or in their own section).